src/schema-validator.js

```diff
@@ -14,68 +14,73 @@ import { dirname, join } from 'path';
 import { fileURLToPath } from 'url';
 import { readFile, readdir } from 'fs/promises';
 
-const SCHEMA_INGESTION_REQUEST = 'ingestion-request';
+export const SCHEMA_INGESTION_REQUEST = 'ingestion-request';
+
+const schemas = {};
+
+export async function getSchema(name, dir) {
+  if (!Object.keys(schemas).length) {
+    let schemasDir = dir;
+    if (!dir) {
+      schemasDir = join(dirname(fileURLToPath(import.meta.url)), 'schemas');
+    }
+    const schemaFiles = await readdir(schemasDir);
+    await Promise.all(
+      schemaFiles.filter((fileName) => fileName.endsWith('.json'))
+        .map(async (fileName) => {
+          schemas[fileName.slice(0, -5)] = JSON.parse(
+            (await readFile(join(schemasDir, fileName))).toString(),
+          );
+        }),
+    );
+  }
+  return schemas[name];
+}
 
 /**
  * Loads schemas from this project in the <code>schemas</code> directory and supports validating
  * objects against the schemas.
  */
 export class SchemaValidator {
-  #loaded = false;
-
-  #schemas = {};
-
-  async #getSchema(name) {
-    if (!this.#loaded) {
-      const schemasDir = join(
-        dirname(fileURLToPath(import.meta.url)),
-        'schemas',
-      );
-      const schemas = await readdir(schemasDir);
-      await Promise.all(
-        schemas
-          .filter((fileName) => fileName.endsWith('.json'))
-          .map(async (fileName) => {
-            const schemaName = fileName.slice(0, -5);
-            const schemaBuf = await readFile(join(schemasDir, fileName));
-            this.#schemas[schemaName] = JSON.parse(schemaBuf.toString());
-          }),
-      );
-      this.#loaded = true;
-    }
-    return this.#schemas[name];
-  }
-
   /**
-   * Validates the <code>ingestionRequest</code> object against the Ingestion Request schema
+   * Validates a request object against the schema specified by <code>schemaName</code>
    * as specified in https://wiki.corp.adobe.com/display/WEM/Ingestor+API+Contract
+   *
+   * Throws <code>Error</code> if request does not match
+   *
    * @see https://wiki.corp.adobe.com/display/WEM/Ingestor+API+Contract?
-   * @param {any} ingestionRequest
+   * @param {any} request
+   * @param {string} schemaName
    * @param {Array<string>} additionalRequiredData
    */
-  async validateIngestionRequest(ingestionRequest, additionalRequiredData) {
-    const schema = await this.#getSchema(SCHEMA_INGESTION_REQUEST);
-    const result = validate(ingestionRequest, schema, {
+  // eslint-disable-next-line class-methods-use-this
+  async validateRequest(request, schemaName, additionalRequiredData) {
```

Comment:
Should this maybe be "validateObject"? Since the thing being validated may not be a request?